src/osd/PrimaryLogPG.cc

```diff
@@ -3466,27 +3467,100 @@ void PrimaryLogPG::cancel_manifest_ops(bool requeue, vector<ceph_tid_t> *tids)
   }
 }
 
-void PrimaryLogPG::refcount_manifest(ObjectContextRef obc, object_locator_t oloc, hobject_t soid,
-                                     SnapContext snapc, bool get, RefCountCallback *cb, uint64_t offset)
+void PrimaryLogPG::dec_refcount_non_intersection(ObjectContextRef obc, const object_info_t& oi, 
+						  const object_ref_delta_t& refs)
+{
+  auto p = refs.begin();
+  while (p!= refs.end()) {
```

Comment:
It starts at begin() and considers every element until end().  Please replace this with a for loop.